packages/expo-updates/android/src/main/java/expo/modules/updates/UpdatesController.java

```diff
@@ -241,7 +249,20 @@ public boolean isEmergencyLaunch() {
   }
 
   /* package */ void resetSelectionPolicyToDefault() {
-    mSelectionPolicy = defaultSelectionPolicy();
+    mSelectionPolicy = mDefaultSelectionPolicy;
+  }
+
+  /* package */ void setDefaultSelectionPolicy(SelectionPolicy selectionPolicy) {
```

Comment:
Instead of methods named `doAAndBAndC()` it's usually ends up better writing `a(); b(); c();` unless there are illegal states between the calls that the interface doesn't want to expose.

Reply:
> Perhaps a way to structure this would be to set mSelectionPolicy to null to indicate that no selection policy has been chosen, and then getSelectionPolicy would set it to mDefaultSelectionPolicy if needed.

That makes the most sense to me; implemented, but is there an idiomatic way to enforce using the getter instead of reading the member field directly inside of UpdatesController? 🤔

Reply:
Not really AFAIK - you just need to be diligent to use the getter, which is often acceptable because the the diligence is scoped to this one class. Tests sometimes can defend against this but I wouldn't say that's reliable because it requires the author to think about writing a test case where they change the default policy before running their main function being tested -- and any author thinking that far ahead probably foresees the need to use the getter.

If it were really important to use the getter -- severe consequences or we find this code repeatedly breaks -- we could define a private helper class that stores the selection policies so that the UpdatesController has to go through the helper class and always use its getter.

Reply:
Gotcha. I'll leave as is for now, maybe we'll refactor this class into kotlin eventually. 😄